### ISSUE TITLE
Don't error on non-existing property when using optional chaining

### ISSUE THREAD

**Issue by kuba-orlik** [microsoft/TypeScript#55079]
# Suggestion

## 🔍 Search Terms

optional chaining "does not exist"
non-existent property with optional chaining

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

Don't throw a TS error when using optional chaining on a property that might not exist in a type.

## 📃 Motivating Example

```ts
type TypeA = {name: string; order: number}
type TypeB = {name: string}

type Type = TypeA | TypeB


const A: Type = Math.random() > 0.5? {name: "A", order: 1} : {name: "B"}

console.log(A?.order || 0)
//            ^ don't throw an error here, it's safe!
```

I can't find any neat workaround for this issue.

## 💻 Use Cases

When you have a union type where one type might have one key and not the other - quite common case, at least for me.

**Comment by nicolo-ribaudo** [comment]
The `?.` there is about `A` being possibly null, not about `A.order`. Since `A` is never null in that example, `A?.order` is exactly the same as `A.order`.
